VideoLocker/src/test/java/org/edx/mobile/discussions/DiscussionTextUtilsTest.java

```diff
@@ -0,0 +1,130 @@
+package org.edx.mobile.discussions;
+
+import android.text.SpannableString;
+import android.text.TextUtils;
+import android.text.style.ClickableSpan;
+import android.widget.TextView;
+
+import org.edx.mobile.discussion.DiscussionTextUtils;
+import org.edx.mobile.discussion.IAuthorData;
+import org.edx.mobile.test.BaseTestCase;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Date;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertTrue;
+
+/**
+ * Created by miankhalid on 25/04/2016.
+ */
+public class DiscussionTextUtilsTest extends BaseTestCase {
+
+    private TextView dummyTextView;
+    private Runnable dummyClickListener;
+
+    @Before
+    public void initValues() {
+        dummyTextView = new TextView(context);
+        dummyClickListener = new Runnable() {
+            @Override
+            public void run() {
+                logger.debug("Author clicked");
+            }
+        };
+    }
+
+    @Test
+    public void test_DuplicateWhitespacesRemoval() {
```

Comment:
Honestly, that name is kind of weird too. Like, it is basically a list of steps, which is essentially what the test itself is, in almost the same level of detail.

Using ``test`` also adds mild redundancy to the encoding. It's pretty easy to forget the @Test annotation and it clearly separates out tests from helpers.

But whether or not we can include the word ``test`` I agree that the name should express the condition being tested.

So this would be more like ``testDuplicatedWhitespaceRemoved`` - so it's a predicate being verified, not a description of a process.